pkg/loki/modules.go

```diff
@@ -490,8 +490,13 @@ func (t *Loki) initStore() (_ services.Service, err error) {
 			// Use AsyncStore to query both ingesters local store and chunk store for store queries.
 			// Only queriers should use the AsyncStore, it should never be used in ingesters.
 			asyncStore = true
-			fallthrough
-		case t.Cfg.isModuleEnabled(IndexGateway) || t.Cfg.isModuleEnabled(Read):
+
+			if t.Cfg.isModuleEnabled(Read) {
+				// we want to use the actual storage when running the index-gateway, so we remove the Addr from the config
+				t.Cfg.StorageConfig.BoltDBShipperConfig.IndexGatewayClientConfig.Disabled = true
```

Comment:
nit: maybe we can extract this to a function. If in the future we make modifications on the `t.Cfg.isModuleEnabled(IndexGateway)` case, they should go inside the if statement as well, but we may skip it unintentionally.

Reply:
I think that's a next step :slightly_smiling_face: